### PR TITLE
Fix/reasons should be array of string

### PR DESCRIPTION
**Pull request by ishiasaka**
#144 
- Move all cancel attributes to subdocument (`order.cancel`)
- Change reason to reasons
- `order.cancel` to be null if order is not canceled.
- Add attributes `ingredients` and `addons` to `order.cancel`

### CODE REVIEW

**Review by lazykern — COMMENTED** [review]
The cancel object in the order response may need to populate the ingredients field and the addons field. In the design, the information about the ingredients and addons that ran out is displayed.

image

edit: FYI im referring to this dto [src/orders/dto/orders.response.dto.ts](https://github.com/meeorder/meeorder-backend/blob/f960d86aa7c7d5341bf9ce69727daae5afc0f8d4/src/orders/dto/orders.response.dto.ts)

**Comment by vjumpkung** [comment]
In this file /features/step-definitions/orders/.ts line 92 ` reason: payload.reason ?? 'Reason',` is it should be `reason: payload.reason?.split(',') ?? [],`

**Review by vjumpkung — COMMENTED** [review]
I think you shold add cancel order with reason and expect to be array of string into cucumber test.

**Comment by lazykern** [comment]
@Vjumpkung I don't get it. You said it should be an array of strings and it is already an array of strings. Why did you change it back to string?

**Comment by vjumpkung** [comment]
I'm sorry for comment not clear enough.
In my branch (#158) I will wait #154 to be merged first.
I saw a step-definition when cancel order still send one string (if it not affect cucumber test it's ok).

**Comment by lazykern** [comment]
Oh it's your PR, I thought you committed into this PR. My bad.

File: src/orders/orders.service.ts [hunk 1]
```diff
@@ -60,36 +61,46 @@ export class OrdersService {
         path: 'menu',
         populate: { path: 'category' },
       })
+      .populate('cancel.ingredients')
+      .populate('cancel.addons')
```
Comment (vjumpkung) [inline src/orders/orders.service.ts:65]: Please edit OrderGetDto class to match a new response.

**Review by lazykern — APPROVED** [review]

**Pull request merged**